.buildkite/pipeline.yml

```diff
@@ -25,24 +25,55 @@ steps:
       memory: "64G"
       image: "docker.elastic.co/ml-dev/ml-linux-build:25"
     commands:
+      - 'du -sh'
+      - 'df -h'
       - 'echo "checkout succeeded"'
       - ".buildkite/scripts/steps/build_and_test.sh"
     depends_on: "check_style"
-    # If changing the key here you will also need to
-    # update the reference in hooks/post-checkout
     key: "build_test_linux-x86_64-RelWithDebInfo"
     artifact_paths:
       - "lib/*/unittest/boost_test_results.junit"
       - "bin/*/unittest/boost_test_results.junit"
     env:
       CMAKE_FLAGS: "-DCMAKE_TOOLCHAIN_FILE=cmake/linux-x86_64.cmake"
+      CPP_CROSS_COMPILE: ""
       RUN_TESTS: "true"
       BOOST_TEST_OUTPUT_FORMAT_FLAGS: "--logger=JUNIT,error,boost_test_results.junit"
     plugins:
       - test-collector#v1.2.0:
           files: "lib/*/unittest/boost_test_results.junit"
           format: "junit"
   
+  - label: "Build & test :cpp: for linux-aarch64-RelWithDebInfo"
+    timeout_in_minutes: 180
+    agents:
+      provider: aws
+      instanceType: m6g.2xlarge
+      imagePrefix: ci-amazonlinux-2-aarch64
+      diskSizeGb: 100
+      diskName: '/dev/xvda'
+    commands:
+      - 'du -sh'
+      - 'df -h'
+      - 'echo "checkout succeeded"'
+      - ".buildkite/scripts/steps/build_and_test.sh || df -h ."
+    depends_on: "check_style"
+    key: "build_test_linux-aarch64-RelWithDebInfo"
+    artifact_paths:
+      - "lib/*/unittest/boost_test_results.junit"
+      - "bin/*/unittest/boost_test_results.junit"
+    env:
+      CMAKE_FLAGS: "-DCMAKE_TOOLCHAIN_FILE=/ml-cpp/cmake/linux-aarch64.cmake"
+      CPP_CROSS_COMPILE: ""
+      RUN_TESTS: "true"
+      BOOST_TEST_OUTPUT_FORMAT_FLAGS: "--logger=JUNIT,error,boost_test_results.junit"
+    plugins:
+      - test-collector#v1.2.0:
+          files: "lib/*/unittest/boost_test_results.junit"
+          format: "junit"
+
+  - wait
```

Comment:
As with the other PR, I don't think it's worth blocking until immediately before the final upload to S3.